lib/image-stream/handlers.js

```diff
@@ -76,6 +77,7 @@ module.exports = {
   'application/gzip': (file, options) => {
     return gzipUncompressedSize.fromFileAsync(file).then((uncompressedSize) => {
       return Bluebird.props({
+        path: file,
```

Comment:
It doesn't sound crazy, but I think we should wait for the streaming functionality to come in before doing such change.